main/src/ca/uwaterloo/flix/language/errors/InstanceError.scala

```diff
@@ -116,4 +116,43 @@ object InstanceError {
     }
   }
 
+  /**
+    * Error indicating the duplicate use of a type variable in an instance type.
+    *
+    * @param tvar the duplicated type.
+    * @param loc  the location where the error occurred.
+    */
+  case class DuplicateTypeParameter(tvar: Type.Var, loc: SourceLocation) extends InstanceError {
+    override def summary: String = "Duplicate type parameter."
+
+    override def message: VirtualTerminal = {
+      val vt = new VirtualTerminal()
+      vt << Line(kind, source.format) << NewLine
+      vt << NewLine
+      vt << Code(loc, s"Duplicate type parameter '${FormatType.formatType(tvar)}'.")
+      vt << NewLine
+      vt << Underline("Tip:") << " Rename one of the instances of the type parameter."
+    }
+  }
+
+  /**
+    * Error indicating a complex instance type.
+    *
+    * @param tpe the complex type.
+    * @param loc the location where the error occurred.
+    */
+  case class ComplexInstanceType(tpe: Type, loc: SourceLocation) extends InstanceError {
```

Comment:
I was half considering whether we should call this something more human friendly, e.g. `NonTypeVariable`, but then I could not find a good name.